kpet/data.py

```diff
@@ -408,6 +459,39 @@ def convert_tree_arches(self):
 
             self.trees[name]["arches"] = list(tree_arches)
 
+    def convert_sets(self):
+        """ Convert each suite / cases sets from a list of regexes
+        to a list of set names matching those regexes
+
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+
+        unassigned = {'unassigned'}
+        error = ("One of '{0}: set:' regexes\n" +
+                 "doesn't match any of the available sets\n" +
+                 "The regex: '{1}'\n" +
+                 "The avaliable sets: " + str(list(self.sets.keys())))
+
+        for suite in self.suites:
+            suite_sets = set()
+            for regex in suite.sets:
+                match = set(filter(regex.fullmatch, self.sets.keys()))
+                if match == set():
+                    raise Invalid(error, "suite", regex.pattern)
+                suite_sets |= match
+            for case in suite.cases:
+                case_sets = suite_sets.copy()
+                for regex in case.sets:
+                    match = set(filter(regex.fullmatch, self.sets.keys()))
+                    if match == set():
+                        raise Invalid(error, "case", regex.pattern)
+                    case_sets |= match
+                if case_sets == set():
+                    case_sets = unassigned
+                case.sets = case_sets.copy()
+            suite.sets = suite_sets.copy()
```

Comment:
We don't need `copy()` in the above two lines, we're abandoning those objects anyway.

Reply:
Yeah, I'm sorry I ran into issues and after the copy above these lines solved it I just slammed it everywhere without thinking to 'make sure'